src/main/java/io/strimzi/kafka/bridge/tracing/OpenTracingHandle.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.jaegertracing.Configuration;
+import io.opentracing.References;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.kafka.TracingKafkaUtils;
+import io.opentracing.contrib.kafka.TracingProducerInterceptor;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+import io.opentracing.propagation.TextMapAdapter;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracer;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.COMPONENT;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.KAFKA_SERVICE;
+
+/**
+ * OpenTracing implementation of TracingHandle.
+ */
+class OpenTracingHandle implements TracingHandle {
+
+    private static Tracer tracer;
+
+    private static Tracer getTracer() {
+        if (tracer == null) {
+            tracer = GlobalTracer.get();
+        }
+        return tracer;
+    }
+
+    static void setCommonTags(Span span, RoutingContext routingContext) {
+        Tags.COMPONENT.set(span, COMPONENT);
+        Tags.PEER_SERVICE.set(span, KAFKA_SERVICE);
+        Tags.HTTP_METHOD.set(span, routingContext.request().method().name());
+        Tags.HTTP_URL.set(span, routingContext.request().uri());
+    }
+
+    @Override
+    public String envServiceName() {
+        return Configuration.JAEGER_SERVICE_NAME;
+    }
+
+    @Override
+    public String serviceName(BridgeConfig config) {
+        return System.getenv(envServiceName());
+    }
+
+    @Override
+    public void initialize() {
+        Tracer tracer = Configuration.fromEnv().getTracer();
+        GlobalTracer.registerIfAbsent(tracer);
+    }
+
+    @Override
+    public <K, V> SpanBuilderHandle<K, V> builder(RoutingContext routingContext, String operationName) {
+        return new OTSpanBuilderHandle<>(getSpanBuilder(routingContext, operationName));
+    }
+
+    @Override
+    public <K, V> SpanHandle<K, V> span(RoutingContext routingContext, String operationName) {
+        Tracer.SpanBuilder spanBuilder = getSpanBuilder(routingContext, operationName);
+        return buildSpan(spanBuilder, routingContext);
+    }
+
+    @SuppressWarnings("rawtypes")
```

Comment:
No need for this AFAICS. The cast can be written `((OTSpanHandle<?,?>) parentSpanHandle)`.